core/domain/topic_services.py

```diff
@@ -1730,8 +1730,10 @@ def get_chapter_counts_in_topic_summaries(
         overdue_chapters_count = 0
         total_chapter_counts = []
         published_chapter_counts = []
-        stories = [story_id_mapping[story_reference.story_id] for
-            story_reference in topic.canonical_story_references]
+        stories = []
+        for story_reference in topic.canonical_story_references:
```

Comment:
Thanks for making this change. 

Clearly one thing we need to do here is to detect that there is a missing story so we can investigate what is going on. Can you have an else condition with a logging.error() that states something like "Topic X has canonical story references [A, B, C] but story reference Y is not a valid story"? Otherwise we are just brushing over the underlying issue which is that the data became invalid somehow.

It's important to do this in a way that results in the logs being seen by the server admin, too. @kevintab95 could you please confirm that logging.error() will be detected by StackDriver?

Reply:
I think it should work. We have integrated the stackdriver client logging with the python's logging module as suggested here: https://stackoverflow.com/a/56048669


https://github.com/oppia/oppia/blob/c8c74fbe0d0b35c3542baa25971941b437a041f3/main.py#L102-L112